### ISSUE TITLE
40 second delay after "server restarted" on Windows

### ISSUE THREAD

**Issue by CaptainN** [meteor/meteor#10367]
After the rebuild process is complete on Windows, I get a 40 second lag between when the terminal prints "Meteor server restarted" and the server is actually accessible from a browser. This is both if I wait for Meteor to refresh itself, and if I intentionally refresh the browser and wait for a response from the server. The same project on Ubuntu in a VM on the same hardware has no lag after "Meteor server restarted", even when testing in the same Windows based browser.

[rebuild.zip](https://github.com/meteor/meteor/files/2654495/rebuild.zip)

I've attached a rebuild profile (also referenced in #5091), but I'm not sure it sheds much light on the post restart lag specifically (though I don't know that I can read that well enough), and I'm not really sure how to debug this further.

I did start with a fresh meteor/react project and try to replicate the problem, but was not able to do so, so it must be something about the way my project is put together. I use a lot of Loadable routes through react-router. I'll see if I can make more of a reduction at some point, but it might just be that the number of files itself is adding this delay.

**Comment by benjamn** [comment]
I wonder if this is because the legacy build is somehow blocking the server, even though that work should be happening in a separate process?

**Comment by CaptainN** [comment]
That raises another question for me - if I save a file while the legacy build is building, does it cancel out or does it finish - or worse, pile up concurrent builds? I have noticed that Meteor does often finish building, the start building again if I save while it's building. It's also difficult to stop the processes if you try in the middle of a rebuild with ctrl+c.

How challenging would it be to attach the legacy build, which runs after "Meteor server restarted" to some sort of additional and continuing output?

**Comment by CaptainN** [comment]
I wonder if there is a difference in the way Windows spins up parallel processes between Home edition and Professional edition? (I'm on Home edition, because why wouldn't Microsoft want to sick up their customers for another $100 after they already bought a Windows machine...)

**Comment by CaptainN** [comment]
I spent some time with this, trying to reproduce the issue on a fresh meteor project. I started with a react project, then added all the packages (atmosphere packages, then npm though without including most of it in the app itself) as I have in one of the projects I see this in, but it didn't have the same effect. There may be something about the specific way I build apps, or it's just a code volume issue. I'll keep investigating when I have the time.
